src/sqlfluff/cli/commands.py

```diff
@@ -299,8 +300,19 @@ def get_config(
                 )
             )
             sys.exit(66)
+
     # Instantiate a config object (filtering out the nulls)
-    overrides = {k: kwargs[k] for k in kwargs if kwargs[k] is not None}
+    def null(key: str, val: Any) -> bool:
+        if val is None:
+            return True
+        if key == "nocolor" and val is False:
+            return True
+        if key == "verbose" and val == 0:
+            return True
+        return False
+
+    overrides = {k: kwargs[k] for k in kwargs if not null(k, kwargs[k])}
```

Comment:
The reason for using `None` as a default value in python is that it is a good way to indicate that a user has not specified an argument versus them explicitly specifying the default value.

If you read your test case, you explicitly pass in `nocolor=False` and receive out `nocolor==True` which is odd behaviour for a function (even if it is doing the correct thing from the CLI perspective):
```python
def test_cli_get_config():
    config = get_config(
        "test/fixtures/config/toml/pyproject.toml",
        True,
        nocolor=False,
        verbose=0,
    )
    assert config.get("nocolor") is True
    assert config.get("verbose") == 2
```

What should happen is if you pass in `nocolor=True` you receive out `True`, if you pass in `nocolor=False` you receive out `False`, and if you pass in `None` (the default) you receive out the value from your config (or the default config if not specified in user config).

Reply:
Can we reset this section to as it was before and instead just simply add `default=None` to the CLI options:
```python
def common_options(f: Callable) -> Callable:
    """Add common options to commands via a decorator.

    These are applied to all of the cli commands.
    """
    f = click.version_option()(f)
    f = click.option(
        "-v",
        "--verbose",
        count=True,
        default=None,
        help=(
            "Verbosity, how detailed should the output be. This is *stackable*, so `-vv`"
            " is more verbose than `-v`. For the most verbose option try `-vvvv` or `-vvvvv`."
        ),
    )(f)
    f = click.option(
        "-n",
        "--nocolor",
        is_flag=True,
        default=None,
        help="No color - if this is set then the output will be without ANSI color codes.",
    )(f)

    return f
```